src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -38,56 +37,730 @@ ActorInit Bg_Dy_Yoseizo_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0B5F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0B75C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0B834.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0B8CC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0BB08.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/BgDyYoseizo_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0BCD8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0BD40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0BE60.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0BF70.s")
+typedef enum GreatFairyAnimation {
+    /* 0 */ GREATFAIRY_ANIM_START_GIVING_UPGRADE,
+    /* 1 */ GREATFAIRY_ANIM_GIVING_UPGRADE,
+    /* 2 */ GREATFAIRY_ANIM_SPIN_LAY_DOWN,
+    /* 3 */ GREATFAIRY_ANIM_LAY_DOWN_TRANSITION,
+    /* 4 */ GREATFAIRY_ANIM_LAYING_DOWN,
+    /* 5 */ GREATFAIRY_ANIM_SHOWING_ITEM,
+    /* 6 */ GREATFAIRY_ANIM_ARMS_FOLDED,
+    /* 7 */ GREATFAIRY_ANIM_CLAPPING,
+    /* 8 */ GREATFAIRY_ANIM_TEACH_SPIN_ATTACK,
+    /* 9 */ GREATFAIRY_ANIM_MAX
+} GreatFairyAnimation;
+
+static AnimationHeader* sAnimations[GREATFAIRY_ANIM_MAX] = {
+    &gGreatFairyStartGivingUpgradeAnim, // GREATFAIRY_ANIM_START_GIVING_UPGRADE
+    &gGreatFairyGivingUpgradeAnim,      // GREATFAIRY_ANIM_GIVING_UPGRADE
+    &gGreatFairySpinLayDownAnim,        // GREATFAIRY_ANIM_SPIN_LAY_DOWN
+    &gGreatFairyLayDownTransitionAnim,  // GREATFAIRY_ANIM_LAY_DOWN_TRANSITION
+    &gGreatFairyLayingDownAnim,         // GREATFAIRY_ANIM_LAYING_DOWN
+    &gGreatFairyShowingItemAnim,        // GREATFAIRY_ANIM_SHOWING_ITEM
+    &gGreatFairyArmsFoldedAnim,         // GREATFAIRY_ANIM_ARMS_FOLDED
+    &gGreatFairyClappingAnim,           // GREATFAIRY_ANIM_CLAPPING
+    &gGreatFairyTeachSpinAttackAnim,    // GREATFAIRY_ANIM_TEACH_SPIN_ATTACK
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dy_Yoseizo/func_80A0C270.s")
+void BgDyYoseizo_Init(Actor* thisx, PlayState* play) {
+    BgDyYoseizo* this = THIS;
+
+    this->unk2EC = this->actor.world.pos.y + 40.0f;
+    this->actor.focus.pos = this->actor.world.pos;
+
+    SkelAnime_InitFlex(play, &this->skelAnime, &gGreatFairySkel, &gGreatFairyShowingItemAnim, this->jointTable,
+                       this->morphTable, GREAT_FAIRY_LIMB_MAX);
+
+    this->actionFunc = func_80A0BB08;
+    Actor_SetScale(&this->actor, 0.0f);
+    this->eyeIndex = 0;
+    this->mouthIndex = 0;
+    this->blinkTimer = 0;
+    this->unk2F8 = 0;
+    GREAT_FAIRY_ROTZ(&this->actor) = 1;
+    this->unk302 = 0;
+}
+
+void BgDyYoseizo_Destroy(Actor* thisx, PlayState* play) {
+}
+
+// Has no visible effect since no segment set for manually-controllable eye textures
```

Comment:
Another extreme nit:
```suggestion
// Has no visible effect since no segment is set for manually-controllable eye textures
```